packages/core/test/oracle/VotingV2.js

```diff
@@ -3037,6 +3049,304 @@ describe("VotingV2", function () {
+    assert.equal((await voting.methods.voterStakes(account1).call()).lastRequestIndexConsidered, 5);
+    assert.equal(
+      (await voting.methods.voterStakes(account1).call()).activeStake,
+      toWei("32000000").add(toWei("108800"))
+    );
+    assert.equal((await voting.methods.voterStakes(account1).call()).unappliedSlash, toWei("108625.92"));
+
+    // Update to 6. Now, we should again see 108625.92 (same math as in previous comment) but both should be applied to
+    // / the active balance.
+    await voting.methods.updateTrackersRange(account1, 6).send({ from: account1 });
+    await voting.methods.updateTrackers(account1).send({ from: account1 });
+    assert.equal((await voting.methods.voterStakes(account1).call()).lastRequestIndexConsidered, 6);
+    assert.equal(
+      (await voting.methods.voterStakes(account1).call()).activeStake,
+      toWei("32000000").add(toWei("108800")).add(toWei("108625.92")).add(toWei("108625.92"))
+    );
+
+    // Update the trackers to 8 now. There should be no further slashing applied and the last request index considered
+    // should not move at all as index7 and 8 are both rolled.
+    await voting.methods.updateTrackersRange(account1, 8).send({ from: account1 });
+    await voting.methods.updateTrackers(account1).send({ from: account1 });
+    assert.equal((await voting.methods.voterStakes(account1).call()).lastRequestIndexConsidered, 6);
+    assert.equal(
+      (await voting.methods.voterStakes(account1).call()).activeStake,
+      toWei("32000000").add(toWei("108800")).add(toWei("108625.92")).add(toWei("108625.92"))
+    );
+
+    // Finally, vote on and slash the last request.
+    baseRequest.roundId = (await voting.methods.getCurrentRoundId().call()).toString();
+    const hash8 = computeVoteHash({ ...baseRequest, price: price, account: account1, time: time + 3 });
+    await voting.methods.commitVote(identifier, time + 3, hash8).send({ from: account1 });
+    await moveToNextPhase(voting, accounts[0]); // Reveal the votes.`
+    await voting.methods.revealVote(identifier, time + 3, price, salt).send({ from: account1 });
+
+    await moveToNextRound(voting, accounts[0]); // Move to the next round.
+
+    // Apply the final slashing trackers. we should see (68e6-108800-108625.92*2)*0.0016 = 108278.317056 added to the
+    // active stake of account1.
+    await voting.methods.updateTrackersRange(account1, 8).send({ from: account1 });
+    await voting.methods.updateTrackers(account1).send({ from: account1 });
+    assert.equal((await voting.methods.voterStakes(account1).call()).lastRequestIndexConsidered, 8);
+    assert.equal(
+      (await voting.methods.voterStakes(account1).call()).activeStake,
+      toWei("32000000").add(toWei("108800")).add(toWei("108625.92")).add(toWei("108625.92")).add(toWei("108278.317056"))
+    );
+  });
+
+  it("Discontinuous rolling interplay with unapplied slashing.", async function () {
```

Comment:
@chrismaree this is the test I created that should pass, but breaks.

The key (I think) is having multiple votes in a round that roll, but having them be discontinuous. In this test, votes 1 and 4 roll, but 2 and 3 do not.